packages/svelte/src/internal/client/proxy.js

```diff
@@ -261,10 +262,20 @@ const state_proxy_handler = {
 		return has;
 	},
 
-	set(target, prop, value) {
+	set(target, prop, value, receiver) {
 		const metadata = target[STATE_SYMBOL];
-		const s = metadata.s.get(prop);
-		if (s !== undefined) set(s, proxy(value, metadata.i, metadata.o));
+		let s = metadata.s.get(prop);
+		// If we haven't yet created a source for this property, we need to ensure
+		// we do so otherwise if we read it later, then the write won't be tracked and
+		// the heuristics of effects will be different vs if we had read the proxied
+		// object property before writing to that property.
+		if (s === undefined && effect_active()) {
+			untrack(() => receiver[prop]);
```

Comment:
This untrack seems to do nothing

Reply:
How do you mean?

Reply:
Yes, but at first glance this property access seems to do nothing, in suggesting a small comment along the lines of 
```suggestion
			// the read creates a signal
			untrack(() => receiver[prop]);
```
(If I'm understanding this correctly)